spark/src/main/java/org/apache/iceberg/spark/source/RowDataRewriter.java

```diff
@@ -98,7 +98,7 @@ public RowDataRewriter(Table table, PartitionSpec spec, boolean caseSensitive,
         task, schema, schema, nameMapping, io.value(), encryptionManager.value(), caseSensitive);
 
     StructType structType = SparkSchemaUtil.convert(schema);
-    SparkAppenderFactory appenderFactory = new SparkAppenderFactory(properties, schema, structType);
+    SparkAppenderFactory appenderFactory = new SparkAppenderFactory(properties, schema, structType, spec);
```

Comment:
Here I use the constructor that has `spec` argument because I believe we will use the `DataWriter` to append records once we switch to the `RollingFileWriter`. https://github.com/apache/iceberg/pull/1818/files#diff-fc9a9fd84d24c607fd85e053b08a559f56dd2dd2a46f1341c528e7a0269f873cR263.